flink-kubernetes/src/main/java/org/apache/flink/kubernetes/kubeclient/decorators/InitJobManagerDecorator.java

```diff
@@ -98,7 +103,7 @@ private Container decorateMainContainer(Container container) {
                 .withImagePullPolicy(kubernetesJobManagerParameters.getImagePullPolicy().name())
                 .withResources(requirements)
                 .withPorts(getContainerPorts())
```

Comment:
Also here are a couple of options which might be configured via the pod template and later overwritten by the default value of a Flink configuration option. I think this goes in the same direction as what Xintong mentioned before. Understanding which parts can be freely configured via the pod template and which parts are read from `flink-conf.yaml` might be confusing to the user. Not sure whether this is feasible or not but maybe we could log whenever we overwrite a pod template configuration with an explicitly configured value from `flink-conf.yaml`. Also it might help to log the resulting pod template which is used for creating the pod.

Reply:
Just like what we have described in the documentation, some fields will always be respected. Flink provides the following list of fields that are planned to be long-term supported: `annotations, labels, imagePullSecrets, nodeSelector, tolerations, env, init container, sidecar container, volumes`. The fields that will be overwritten are listed in the tables.

Adding a sanity check for the pod template in one method does not make sense to me. Because it will also be a burden to maintain especially when we change the internal overwrite logic or introduce new config options. I will try to figure out whether it is feasible to log whenever we overwrite a pod template configuration in decorators.

Currently, we already log the result pod meta and spec in the `Fabric8FlinkKubeClient` when creating JobManager and TaskManager.